cmake/modules/SwiftUtils.cmake

```diff
@@ -1,136 +1,174 @@
+# Module import guard
+if(DEFINED SWIFT_UTILS_MODULE_LOADED)
+  return()
+endif()
+set(SWIFT_UTILS_MODULE_LOADED TRUE)
+
+
 include(CMakeParseArguments)
 
+
 function(precondition var)
-  cmake_parse_arguments(
-    PRECONDITION # prefix
-    "NEGATE" # options
-    "MESSAGE" # single-value args
-    "" # multi-value args
+  set(options NEGATE)
+  set(single_value_args MESSAGE)
+  set(multi_value_args)
+
+  cmake_parse_arguments(PRE
+    "${options}"
+    "${single_value_args}"
+    "${multi_value_args}"
     ${ARGN})
 
-  if (PRECONDITION_NEGATE)
-    if (${var})
-      if (PRECONDITION_MESSAGE)
-        message(FATAL_ERROR "Error! ${PRECONDITION_MESSAGE}")
+  if(PRE_NEGATE)
+    if(${var})
+      if(PRE_MESSAGE)
+        message(FATAL_ERROR "Error! ${PRE_MESSAGE}")
       else()
-        message(FATAL_ERROR "Error! Variable ${var} is true or not empty. The value of ${var} is ${${var}}.")
+        message(FATAL_ERROR
+          "Error! Variable ${var} is true or not empty. "
+          "The value of ${var} is ${${var}}.")
       endif()
     endif()
   else()
```

Comment:
Why not use an early return here? We are using a late enough cmake to use return().